lib/inspec/plugin/v1/plugin_types/resource.rb

```diff
@@ -83,11 +83,14 @@ def initialize(backend, name, *args)
           @__resource_name__ = name
 
           # check resource supports
-          supported = true
-          supported = check_supports unless @supports.nil?
+          supported = @supports ? check_supports : true
           test_backend = defined?(Train::Transports::Mock::Connection) && backend.backend.class == Train::Transports::Mock::Connection
-          # do not return if we are supported, or for tests
-          return unless supported || test_backend
+          # raise if we are supported, or for tests
+          unless supported || test_backend
+            msg = "Unsupported resource/backend combination: %s / %s. Exiting." %
+              [name, backend.platform.name]
+            raise ArgumentError, msg
```

Comment:
Here's an idea, which would at least improve messaging - create a custom exception class (or we may have a suitable one already - see lib/exceptions or lib/errors, I know we have two). Perhaps ResourceUnsupportedError? Then add a `rescue` matching that new exception class in the control evaluator - https://github.com/inspec/inspec/blob/954e6c0ff98113ed41e95b50f5a6651a30572c8c/lib/inspec/rule.rb#L54 . It make a special case of this, but would give you the chance to handle the outcome with a little more finesse.